helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -1227,6 +1299,19 @@ private void pushToEventQueues(ClusterEventType eventType, NotificationContext c
     for (Map.Entry<String, Object> attr : eventAttributes.entrySet()) {
       event.addAttribute(attr.getKey(), attr.getValue());
     }
+
+    // Management mode event will force management mode pipeline.
+    if (ClusterEventType.ManagementModePipeline.equals(eventType)
+        || _managementControllerDataProvider.shouldRunManagementPipeline()) {
```

Comment:
Same here, better not use the object reference. We have the cache dedicated in event. Let's bound the operation in the function scope.